### PR TITLE
integration-tests: Remove the integration test image

### LINKED ISSUE #944: [RFE] Decide what to do with the integration test image
## Current situation

To run integration tests, we first build a docker image which contains a compiled binary containing our integration tests.
On release, this image is then pushed as [`inspektor-gadget-test`](https://github.com/kinvolk/inspektor-gadget/pkgs/container/inspektor-gadget-test).

## Impact

To keep our sanity, the integration tests jobs are factorized running in the [`run-integration-tests action`](https://github.com/kinvolk/inspektor-gadget/blob/3af35cb183c576b7a62f3887157e6b7701b5725a/.github/actions/run-integration-tests/action.yml).
As a consequence, this will mean the above image will built for each call to this action but also pushed on release.

## Ideal future situation

The goal here is to discuss what should we do about this particular image.
Should we stop publishing it somewhere? Should we only publish it once when all the integration tests suceeded?

### PR DESCRIPTION
**Pull request by mqasimsarfraz**
* The initial idea for having a docker image was to allow running IG tests on a specific Kubernetes environment without having to set up a CI. But it ended up not being used.
* It also removes the integration module (not needed anymore) removing the multi-module go dependency.

Fixes #944 

### Testing Done:
- Clean up workflow: https://github.com/mqasimsarfraz/inspektor-gadget/actions/runs/3091476571/jobs/5002582050
- Tag workflow: https://github.com/mqasimsarfraz/inspektor-gadget/actions/runs/3090761608

### CODE REVIEW

File: .github/actions/run-integration-tests/action.yml
Comment (mauriciovasquezbernal) [inline .github/actions/run-integration-tests/action.yml:28]: Where is the PID variable initialized?

File: .github/actions/run-integration-tests/action.yml
Comment (mauriciovasquezbernal) [inline .github/actions/run-integration-tests/action.yml:44]: Maybe we can add a comment with a link to http://mywiki.wooledge.org/SignalTrap#When_is_the_signal_handled.3F to avoid confusion later on on this.

**Review by mauriciovasquezbernal — COMMENTED** [review]
Looks very good. Some questions regarding the clean up procedure.

File: .github/actions/run-integration-tests/action.yml
Comment (mqasimsarfraz) [reply .github/actions/run-integration-tests/action.yml:28]: Thanks for spotting. I was suppose to use `$PIDS` here.

File: Makefile
Comment (mqasimsarfraz) [reply Makefile:168]: I will add it back. Make sense to keep it to allow different image.

File: .github/actions/run-integration-tests/action.yml
Comment (blanquicet) [inline .github/actions/run-integration-tests/action.yml:27]: Did you try this using the cancellation?

File: .github/workflows/inspektor-gadget.yml
Comment (blanquicet) [inline .github/workflows/inspektor-gadget.yml:473]: I don't see why it shouldn't work, but could you please try a merge and a release (I usually comment the release step to do it) on a fork to be sure it is working properly?

**Review by blanquicet — COMMENTED** [review]
Thanks for working on this and opening the PR that quickly! This change is definitely what we need in terms of simplicity (removing the multi-modules env).

I added some comments, and I think it would be great if you were able to test it in a fork. Thanks!

File: .github/actions/run-integration-tests/action.yml
Comment (mqasimsarfraz) [reply .github/actions/run-integration-tests/action.yml:11]: Just so that the binary downloaded or built locally are at the same location as expected [here](https://github.com/kinvolk/inspektor-gadget/blob/main/Makefile#L165).

File: Makefile
Comment (mqasimsarfraz) [reply Makefile:168]: This can now be handled via `INTEGRATION_TESTS_PARAMS` like [here](https://github.com/kinvolk/inspektor-gadget/pull/955/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R165).

File: .github/actions/run-integration-tests/action.yml
Comment (mqasimsarfraz) [reply .github/actions/run-integration-tests/action.yml:4]: Probably using `KUBERNETES_DISTRIBUTION` as input is more clear. Also do you have use-case in mind for passing `container_repo` and `image_tag` in CI? If yes then we should pass them separately.

File: Makefile
Comment (blanquicet) [reply Makefile:168]: I think `-image` should be managed outside `INTEGRATION_TESTS_PARAMS` as it can be configured using `CONTAINER_REPO` and `IMAGE_TAG`. My idea with `INTEGRATION_TESTS_PARAMS` is more for debugging, when one wants to run just some tests `-run`, or multiple time `-count` or do not deploy the IG or SOP.

cc @eiffel-fl

Comment (mqasimsarfraz) [reply Makefile:168]: Yeah, makes sense. I will update the PR.

File: .github/actions/run-integration-tests/action.yml
Comment (mqasimsarfraz) [reply .github/actions/run-integration-tests/action.yml]: Yes, just tested it [here](https://github.com/mqasimsarfraz/inspektor-gadget/actions/runs/3090481174/jobs/5000096159). Thanks for suggesting I was missing a `;` which would have broken the cleanup. :smile:

File: .github/workflows/inspektor-gadget.yml
Comment (mqasimsarfraz) [reply .github/workflows/inspektor-gadget.yml:473]: A green CI run with tag can be found [here](https://github.com/mqasimsarfraz/inspektor-gadget/actions/runs/3090761608).

**Comment by mqasimsarfraz** [comment]
Thanks everyone for the reviews. I have updated the PR to address all the comments. Would be great if you have another look!

File: .github/actions/run-integration-tests/action.yml
Comment (mqasimsarfraz) [reply .github/actions/run-integration-tests/action.yml]: I don't see any reason why it shouldn't work. `;\` is bit error prone as well. I will try without it.

File: .github/workflows/inspektor-gadget.yml
Comment (blanquicet) [inline .github/workflows/inspektor-gadget.yml:211]: Was it missing?

File: .github/actions/run-integration-tests/action.yml
Comment (blanquicet) [reply .github/actions/run-integration-tests/action.yml:37]: I remember it didn't work, but I didn't investigate it too much 🙈

**Review by blanquicet — APPROVED** [review]
After verifying that the cleanup is still working properly after removing all the `;\`, I think it is ready to go 🎉!

Please wait for another approval before merging, I would like someone else to verifies we didn't miss anything 🙂

**Review by mauriciovasquezbernal — APPROVED** [review]
LGTM from a code inspection. I like that it's cleaning those integration tests a lot.

File: .github/workflows/inspektor-gadget.yml
Comment (mqasimsarfraz) [reply .github/workflows/inspektor-gadget.yml:211]: Yep.

File: .github/actions/run-integration-tests/action.yml
Comment (blanquicet) [inline .github/actions/run-integration-tests/action.yml:41]: Could you please explain why to kill `pidof integration.test` but wait for `jobs -p`?

Comment (mqasimsarfraz) [reply .github/actions/run-integration-tests/action.yml]: Thanks for spotting, I changed to `pidof integration.test` in all the places since that is more clear.

Comment (mqasimsarfraz) [reply .github/actions/run-integration-tests/action.yml]: Also, the cleanup function is [working fine](https://github.com/mqasimsarfraz/inspektor-gadget/actions/runs/3091476571/jobs/5002582050) without  `;\`

Comment (blanquicet) [reply .github/actions/run-integration-tests/action.yml]: If you tested the cleanup after you switched to use `pidof integration.test`, we are ready to go 🚀

Comment (mqasimsarfraz) [reply .github/actions/run-integration-tests/action.yml]: Found minor [issue](https://github.com/mqasimsarfraz/inspektor-gadget/actions/runs/3093036278/jobs/5005205344) with the approach. So I am going to [stick](https://github.com/mqasimsarfraz/inspektor-gadget/actions/runs/3093714084/jobs/5006572610) with what I had before, added comment why we need it. We are ready for merge if you think it is looking good! :partying_face:

Comment (blanquicet) [reply .github/actions/run-integration-tests/action.yml]: The comment is clear, so is enough for now. We can investigate later if killing and waiting for the same process is possible. It would make the code clear.

**Review by blanquicet — APPROVED** [review]

**Pull request merged**
